data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/SearchAccessorStrategy.java

```diff
@@ -189,14 +210,58 @@ private boolean versionSupportsPointInTime(final String distribution, final Stri
         if (Objects.nonNull(infoResponseOpenSearch)) {
-            return Pair.of(infoResponseOpenSearch.version().distribution(), infoResponseOpenSearch.version().number());
+            final String distribution = infoResponseOpenSearch.version().distribution();
+            final String versionNumber = infoResponseOpenSearch.version().number();
+            if (Objects.nonNull(distribution) && Objects.nonNull(versionNumber)) {
+                LOG.info("Detected OpenSearch distribution '{}' version '{}' from API response.", distribution, versionNumber);
+                return Pair.of(distribution, versionNumber);
+            }
+            LOG.warn("Distribution or version number is null in OpenSearch API response. Proceeding to check Elasticsearch or default.");
         }
 
-        try {
-            final co.elastic.clients.elasticsearch.core.InfoResponse infoResponseElasticsearch = elasticsearchClientRefresher.get().info();
-            return Pair.of(ELASTICSEARCH_DISTRIBUTION + "-" + infoResponseElasticsearch.version().buildFlavor(), infoResponseElasticsearch.version().number());
-        } catch (final Exception e) {
-            throw new RuntimeException("Unable to call info API using the elasticsearch client", e);
+        if (elasticsearchClientRefresher != null) {
+            try {
+                final ElasticsearchClient elasticsearchClient = elasticsearchClientRefresher.get();
+                if (elasticsearchClient == null) {
+                    LOG.info("ElasticsearchClient from refresher is null. Cannot attempt Elasticsearch .info() call. Proceeding to default.");
+                } else {
+                    final co.elastic.clients.elasticsearch.core.InfoResponse infoResponseElasticsearch = elasticsearchClient.info();
+                    final String esBuildFlavor = infoResponseElasticsearch.version().buildFlavor();
+                    final String esVersionNumber = infoResponseElasticsearch.version().number();
+
+                    if (Objects.nonNull(esVersionNumber)) {
+                        String effectiveDistribution = ELASTICSEARCH_DISTRIBUTION;
+                        LOG.info("Detected Elasticsearch (flavor '{}') version '{}' from API response.",
+                                Objects.toString(esBuildFlavor, "unknown"), esVersionNumber);
+                        if (ELASTICSEARCH_OSS_BUILD_FLAVOR.equalsIgnoreCase(esBuildFlavor)) {
+                            effectiveDistribution = ELASTICSEARCH_DISTRIBUTION + "-" + ELASTICSEARCH_OSS_BUILD_FLAVOR;
+                        }
+                        return Pair.of(effectiveDistribution, esVersionNumber);
+                    }
+                    LOG.warn("Version number is null in Elasticsearch API response. Proceeding to default.");
+                }
+            } catch (final Exception e) {
+                LOG.warn("Failed to get cluster info using Elasticsearch client. Will proceed with default behavior.", e);
```

Comment:
This log could maybe indicate that the default behavior is to assume that it is OpenSearch, although you do have that log later on